src/components/year_month.rs

```diff
@@ -61,6 +68,80 @@ impl<C: CalendarProtocol> YearMonth<C> {
     pub fn calendar(&self) -> &CalendarSlot<C> {
         &self.calendar
     }
+
+    /// Returns the Calendar value.
+    #[inline]
+    #[must_use]
+    pub fn in_leap_year(&self) -> bool {
+        utils::mathematical_in_leap_year(utils::epoch_time_for_year(self.iso.year)) == 1
+    }
+}
+
+// Contextual Methods
+impl<C: CalendarProtocol> YearMonth<C> {
+    pub fn contextual_get_days_in_year(
+        this: &C::YearMonth,
+        context: &mut C::Context,
+    ) -> TemporalResult<u16> {
+        this.get_calendar()
+            .days_in_year(&CalendarDateLike::YearMonth(this.clone()), context)
+    }
+
+    pub fn contextual_get_days_in_month(
+        this: &C::YearMonth,
+        context: &mut C::Context,
+    ) -> TemporalResult<u16> {
+        this.get_calendar()
+            .days_in_month(&CalendarDateLike::YearMonth(this.clone()), context)
+    }
+
+    pub fn contextual_month_code(
+        this: &C::YearMonth,
+        context: &mut C::Context,
+    ) -> TemporalResult<TinyAsciiStr<4>> {
+        this.get_calendar()
+            .month_code(&CalendarDateLike::YearMonth(this.clone()), context)
+    }
+
+    pub fn contextual_get_months_in_year(
+        this: &C::YearMonth,
+        context: &mut C::Context,
+    ) -> TemporalResult<u16> {
+        this.get_calendar()
+            .months_in_year(&CalendarDateLike::YearMonth(this.clone()), context)
+    }
+
+    pub fn contextual_add_or_subtract_duration(
```

Comment:
issue: this method should probably be `pub(crate)` at most.

`AddOrSubtractDuration` should probably be an internal abstract method. I think ideally, the public API here should be `add` & `subtract`. This would also save adding the `DurationOperation` enum.